src/main/scala/ai/privado/dataflow/Dataflow.scala

```diff
@@ -208,6 +221,25 @@ class Dataflow(cpg: Cpg, statsRecorder: StatsRecorder) {
     }
   }
 
+  private def addOriginalSourceToDataflowPath(dataflow: Path): List[Path] = {
+    val dataflowIterator = dataflow.elements
+
+    val originalSourcesForHeadElement =
+      if (dataflowIterator.nonEmpty)
+        dataflowIterator.head.originalSourceOut
+          .map(c => c.asInstanceOf[AstNode])
```

Comment:
Why do we have to typecast this every time, can this be not handled as part of the `originalSourceOut` query?